rotkehlchen/tasks/calendar.py

```diff
@@ -75,123 +78,203 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
+            if self.db_calendar.count_reminder_entries(
+                    event_id=(entry_id := calendar_entry.identifier),
             ) > 0:  # already has a reminder entry
-                continue  # we don't add any new automatic reminders
+                return None  # we don't add any new automatic reminders
 
-        if (
-            ens_name not in ens_to_event or
-            ens_expires > ens_to_event[ens_name][1].extra_data['expires']  # type: ignore[index]  # extra_data is not None, checked above
-        ):  # insert mapping for the latest expiry timestamp
-            ens_to_event[ens_name] = (calendar_entry_id, ens_event)
-
-    _, failed_to_add = db_calendar.create_reminder_entries(reminders=[
-        ReminderEntry(
-            identifier=calendar_identifier,  # this is only used for logging below, it's auto generated in db  # noqa: E501
-            event_id=calendar_identifier,
-            secs_before=secs_before,
+        return entry_id
+
+    def maybe_create_reminders(self, calendar_identifiers: list[int], secs_befores: list[int], error_msg: str) -> None:  # noqa: E501
+        _, failed_to_add = self.db_calendar.create_reminder_entries(reminders=[
+            ReminderEntry(
+                identifier=calendar_identifier,  # this is only used for logging below, it's auto generated in db  # noqa: E501
+                event_id=calendar_identifier,
+                secs_before=secs_before,
+            )
+            for calendar_identifier in calendar_identifiers
+            for secs_before in secs_befores
+        ])
+
+        if len(failed_to_add) == 0:
+            return
+
+        log.error(  # failed_to_add is a list of calendar_identifier that were passed as identifiers of reminder entry above  # noqa: E501
+            f"""{error_msg} for {', '.join([
+                entry.name for entry in self.db_calendar.query_calendar_entry(
+                    CalendarFilterQuery.make(identifiers=failed_to_add)
+                )['entries']
+            ])}""",
         )
-        for calendar_identifier, _ in ens_to_event.values()
-        for secs_before in (WEEK_IN_SECONDS, DAY_IN_SECONDS)
-    ])
+
+    def maybe_create_ens_reminders(self) -> None:
+        """Check ENS registration and renewal history events and create reminders if needed."""
+        ens_events = self.get_history_events(
+            event_types=[
+                (HistoryEventType.TRADE, HistoryEventSubType.SPEND),
+                (HistoryEventType.RENEW, HistoryEventSubType.NONE),
+            ],
+            counterparty=CPT_ENS,
+        )
+
+        if len(ens_events) == 0:
+            return
+
+        ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+        for ens_event in ens_events:
+            assert ens_event.extra_data is not None
+            if (
+                (ens_name := ens_event.extra_data.get('name')) is None or
+                (ens_expires := ens_event.extra_data.get('expires')) is None
+            ):
```

Comment:
remove this assertion. As i said in the other comment it's possible to then get here with extra_data being `None` due to the fact until before this PR is merged then the corresponding events had no extra data.

```suggestion
            if (
                not (extra_data := ens_event.extra_data or {}) or
                (ens_name := ens_event.extra_data.get('name')) is None or
                (ens_expires := ens_event.extra_data.get('expires')) is None
            ):
```